opslevel/resource_opslevel_service.go

```diff
@@ -1,5 +1,333 @@
 package opslevel
 
+import (
+	"context"
+	"fmt"
+	"slices"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/opslevel/opslevel-go/v2024"
+)
+
+var _ resource.ResourceWithConfigure = &ServiceResource{}
+
+var _ resource.ResourceWithImportState = &ServiceResource{}
+
+func NewServiceResource() resource.Resource {
+	return &ServiceResource{}
+}
+
+// ServiceResource defines the resource implementation.
+type ServiceResource struct {
+	CommonResourceClient
+}
+
+// ServiceResourceModel describes the Service managed resource.
+type ServiceResourceModel struct {
+	Aliases                    types.List   `tfsdk:"aliases"`
+	ApiDocumentPath            types.String `tfsdk:"api_document_path"`
+	Description                types.String `tfsdk:"description"`
+	Framework                  types.String `tfsdk:"framework"`
+	Id                         types.String `tfsdk:"id"`
+	Language                   types.String `tfsdk:"language"`
+	LastUpdated                types.String `tfsdk:"last_updated"`
+	LifecycleAlias             types.String `tfsdk:"lifecycle_alias"`
+	Name                       types.String `tfsdk:"name"`
+	Owner                      types.String `tfsdk:"owner"`
+	OwnerId                    types.String `tfsdk:"owner_id"`
+	PreferredApiDocumentSource types.String `tfsdk:"preferred_api_document_source"`
+	Product                    types.String `tfsdk:"product"`
+	Tags                       types.List   `tfsdk:"tags"`
+	TierAlias                  types.String `tfsdk:"tier_alias"`
+}
+
+func NewServiceResourceModel(ctx context.Context, service opslevel.Service) (ServiceResourceModel, diag.Diagnostics) {
+	var diags diag.Diagnostics
+	serviceResourceModel := ServiceResourceModel{
+		ApiDocumentPath: OptionalStringValue(service.ApiDocumentPath),
+		Description:     OptionalStringValue(service.Description),
+		Framework:       OptionalStringValue(service.Framework),
+		Id:              ComputedStringValue(string(service.Id)),
+		Language:        OptionalStringValue(service.Language),
+		LifecycleAlias:  OptionalStringValue(service.Lifecycle.Alias),
+		Name:            RequiredStringValue(service.Name),
+		Owner:           OptionalStringValue(service.Owner.Alias),
```

Comment:
I noticed for this field that the description says `"id or alias may be used"` and this function always populates this field with the `alias`. 

What if the user decided to use an ID here? Would that result in error messages/unnecessary updates? 

Same thing happens in other resources where a field allows an alias OR ID.

Reply:
@davidbloss to address this you can add something like:

https://github.com/OpsLevel/terraform-provider-opslevel/pull/247/files#diff-c294bb8f5a3b817c0ee0a0e6054a395f6fde4d7cd4a9f8d55ff29700db4c4054R71-R79